support/cmake/cxx14.cmake

```diff
@@ -1,43 +1,50 @@
 # C++14 feature support detection
 
-include(CheckCXXSourceCompiles)
 include(CheckCXXCompilerFlag)
+macro (fmt_check_cxx_compiler_flag _FLAG _RESULT)
```

Comment:
Let's make it a function instead of a macro unless there is a strong reason not to and use normal naming convention, i.e. `flag` and `result` for arguments.